### PR TITLE
Fix/kminimization

### PR DESCRIPTION
**Pull request by nmaeder**
This PR fixes/changes a few different things regarding the ETK Minimization.

### 12, 13 and 14 Distance Constraints

Fixes a bug, where the wrong atom pair indices were stored when constraining bonds, angles and dihedrals.
This led to atom pairs that had more than one distance constraint applied to them during ETK minimization.

Also inhibit adding a 1-3 Distance Constraint for which the central atom would be in an improper torsion contribution.

### Lower SP Linearity Force Constant

Lowering the force Constant for Linear SP Atoms from 100 to 1. The high force constant led to a distortion of bond lengths
when paired with ET terms in cases where the ET terms where only able to pull through an SP center by changing a lot
of bond lenghts in the conformer.

### Fix ETDG Distance Constraints

Fix a bug where when only using ETDG, the distance constraints provided were not enforced.

### Varia

- Some range based for loops introduced where outdated for loops were used.
- Uniquifying `DistGeom::Construct3DForceField` and `DistGeom::ConstructPlain3DForceField` where the code was doing the same (can possibly be refactored to prevent code duplication at some point)
- Add a SP Linearity contribution to `DistGeom::construct3DImproperForceField` to make sure, linarity of SP centers is also checked at the end of the ETKDG workflow.
- Adapt tests.

### CODE REVIEW

**Review by greglandrum — CHANGES_REQUESTED** [review]
I made a few small suggestions here and also did a PR against your branch with a few others that were easier to make as code

(plus I wanted to test them ;-))

File: Code/GraphMol/DistGeomHelpers/catch_tests.cpp [hunk 1]
```diff
@@ -593,55 +594,30 @@ TEST_CASE("double bond stereo not honored in conformer generator") {
   }
 }
 
-TEST_CASE("tracking failure causes"){SECTION("basics"){
-    auto mol =
-        "C=CC1=C(N)Oc2cc1c(-c1cc(C(C)O)cc(=O)cc1C1NCC(=O)N1)c(OC)c2OC"_smiles;
+TEST_CASE("tracking failure causes"){
```
[the remark this answers is not in the document]
Comment (nmaeder) [reply Code/GraphMol/DistGeomHelpers/catch_tests.cpp:597]: I think it is also messed up in the master branch. Running clang-format gives me exactly this, but i can double check.

File: Code/GraphMol/DistGeomHelpers/testDgeomHelpers.cpp [hunk 2]
```diff
@@ -2346,10 +2346,6 @@ void testGithub3667() {
 void testForceTransAmides() {
   auto mol = "CC(=O)NC"_smiles;
   TEST_ASSERT(mol);
-  bool updateLabel = true;
-  bool takeOwnership = true;
-  mol->addAtom(new Atom(1), updateLabel, takeOwnership);
-  mol->addBond(3, 5, Bond::BondType::SINGLE);
```
[the remark this answers is not in the document]
Comment (nmaeder) [reply Code/GraphMol/DistGeomHelpers/testDgeomHelpers.cpp:2352]: i reverted to the old code.

File: Code/GraphMol/DistGeomHelpers/catch_tests.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (nmaeder) [reply Code/GraphMol/DistGeomHelpers/catch_tests.cpp:597]: Seems to be a bug in the vs code clang-format extension, running it from the cli fixes it, but as soon as i format it from within vscode, it gets funky again. Will push a cli formatted version

**Comment by nmaeder** [comment]
Changed code according to the review

File: Code/GraphMol/DistGeomHelpers/catch_tests.cpp [hunk 3]
```diff
@@ -593,55 +594,30 @@ TEST_CASE("double bond stereo not honored in conformer generator") {
...
+    SECTION("basics"){auto mol = "O=c2cc3CCc1ccc(cc1Br)CCc2c(O)c3=O"_smiles;
```
Comment (greglandrum) [inline Code/GraphMol/DistGeomHelpers/catch_tests.cpp]: this formatting is still screwed up, but it's not the end of the world

File: Code/GraphMol/DistGeomHelpers/catch_tests.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (greglandrum) [reply Code/GraphMol/DistGeomHelpers/catch_tests.cpp]: I also noticed that I was getting weird formatting. Installing a recent version of clang-format with conda (this gives me v18) and then pointing VS Code at that fixes that.

**Pull request merged**